Source/SwiftLintFramework/Extensions/NSRegularExpression+SwiftLint.swift

```diff
@@ -13,22 +13,32 @@ public typealias NSRegularExpression = RegularExpression
 public typealias NSTextCheckingResult = TextCheckingResult
 #endif
 
-private var regexCache = [String: NSRegularExpression]()
+private var regexCache = [RegexCacheKey: NSRegularExpression]()
+
+private struct RegexCacheKey: Hashable {
```

Comment:
I thought it was easier to just create a type to wrap the key then implementing a full multimap 😅